### ISSUE TITLE
How to tune pressure_advance_lookahead_time and square_corner_velocity?

### ISSUE THREAD

**Issue by TERRAOperative** [KevinOConnor/klipper#1845]
I've been tuning a new klipper install on my 3D printer and I am getting ghosting/ringing after corners.

I have been looking through the documentation on how to tune the 'pressure_advance_lookahead_time' and 'square_corner_velocity' parameters, but I can't find anything.

Can anyone please let me know how to tune these parameters properly?
Or if there is another parameter I should be tuning to remove the ringing? It seems to print perfectly otherwise.

**Comment by klipper-gitissuebot** [comment]
Hi @TERRAOperative,

It did not look like there was a Klipper log file attached to this ticket. The log file has been engineered to answer common questions the Klipper developers have about the software and its environment (software version, hardware type, configuration, event timing, and hundreds of other questions).

Unfortunately, too many people have opened tickets without providing the log. That consumes developer time; time that would be better spent enhancing the software. If this ticket references an event that has occurred while running the software then the Klipper log must be attached to this ticket. Otherwise, this ticket will be automatically closed in a few days.

For information on obtaining the Klipper log file see: https://github.com/KevinOConnor/klipper/blob/master/docs/Contact.md

The log can still be attached to this ticket - just add a comment and attach the log to that comment.

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being.

**Comment by TERRAOperative** [comment]
[klippy.log](https://github.com/KevinOConnor/klipper/files/3440081/klippy.log)

**Comment by Xono** [comment]
@TERRAOperative 

The steps for pressure advance tuning are detailed in [docs/Pressure_Advance.md](https://github.com/KevinOConnor/klipper/blob/master/docs/Pressure_Advance.md). This document also contains a note that the `pressure_advance_lookahead_time` value should be set to 0 while tuning, and be returned to its default of 0.010 for normal use - though it can be adjusted manually if needed. The `square_corner_velocity` field is described in [docs/kinematics.md](https://github.com/KevinOConnor/klipper/blob/master/docs/Kinematics.md) as being used to calculate the speed through different sharp angles, though not how different values affect the print itself. 

It is detailed that a restart command will reset both of these to their defaults, meaning you generally shouldn't need to alter these manually - the extruder's `pressure_advance` value is the only one you should need to be tweaking by hand yourself, as described in the steps of the Pressure Advance document.

**Comment by BlackStump** [comment]
try reducing your acceleration, especially if it is a bed slinger ie Ender3
your log has 
max_accel = 3000
Try
max_accel = 1000

**Comment by TERRAOperative** [comment]
@Xono I have done the Pressure_Advance tuning which worked well for the corners, but I didn't see the kinematics document yet, thanks for the link. :)

@BlackStump I tried a test print at max_accel = 2000 and the ringing is less now, so that is on the right track.
My printer is an extensively modified Wanhao Duplicator 6, so I was hoping it would be less prone to this sort of thing, having a full Ultimaker type enclosure frame.
I wonder if the stepper motor dampers I fitted are causing a little bit of compliance and causing the ringing?...

More tweaking ahoy. :D
